### DISCUSSION TITLE
Differences between JtlReporter and JMeter HTML Report (Scenario Trends vs. Statistics Table)

### DISCUSSION THREAD

**Discussion by MoritzSalein** [ludeknovy/jtl-reporter#113] [Q&A] [answered]
I compare the summary results between JMeter HTML Report and JtlReporter and I found following differences between Statistics table and scenario trends:
1. Response Time P90
2. Network Bytes

The same differences are on the Overview panel of the test run.

Which report is incorrect?

![Network](https://user-images.githubusercontent.com/83296020/151504030-4610fee0-1345-47e0-9858-76db2d833575.png)
![ResponseTimeP90](https://user-images.githubusercontent.com/83296020/151504035-b2d1f5c5-e361-451f-a42b-2b1818529484.png)
![Overview](https://user-images.githubusercontent.com/83296020/151504036-0e77edb8-c000-4b4d-9f49-a6fd86f018af.png)
.

**Comment by ludeknovy** [comment]
Hi @MoritzSalein! 
Regarding the differences in P90 - I think JtlReporter calculates it only for successful requests. In both examples you provided, I can see there was a small number of failed requests and that could make the difference.

Regarding the differences in - this looks correct in the report detail. But please note they use different units. 
If you convert ~1883KB/sec (JMeter HTML report) to Mbps, it makes ~15. https://www.convertunits.com/from/KB/s/to/Mbps
But it definitely looks off in the scenario chart.

**Reply by MoritzSalein** [reply]
Hi @ludeknovy,

I understand your explanation to the P90 value. But I ran the same scenario again later without errors and I got a similiar difference. Look at my new screenshot.

The small difference of calculating the network bytes in the report details is acceptable to me. Of course, I had a little confusion of the units.

![Rerun_P90](https://user-images.githubusercontent.com/83296020/151521994-59e04660-8133-4a29-b846-e5e1a1519b8b.png)

**Reply by ludeknovy** [reply]
1) I checked the code and for the calculation I use PSQL **percentile_cont**  function - *continuous percentile: returns a value corresponding to the specified fraction in the ordering, interpolating between adjacent input items if needed.* see: https://www.postgresql.org/docs/14/functions-aggregate.html

2) If I recall correctly, the unit transformation is done in UI, so I could make a setting where you could set-up the units per scenario. If that would help.

**Reply by MoritzSalein** [reply]
to 1. ok, I don't know how the JMeter HTML Report works, but it must be different. I try to ask the JMeter project how they calculate it. 
to 2. yes, it will be a nice feature to set up the units in the interface.

**Comment by MoritzSalein** [comment, marked as answer]
@ludeknovy Perhaps the difference in the scenario trend graph could be the same issue as reported in https://github.com/ludeknovy/jtl-reporter/issues/98

**Reply by ludeknovy** [reply]
Yes, it looks like. I will check that again. What version do you use?

**Reply by MoritzSalein** [reply]
I use a cloned version 4.0 from the 9th December. In the docker-compose.yml I found the version 2.1.
But I don' t know where I can find the build version.

**Reply by ludeknovy** [reply]
Thanks! That's enough infos for me! Just wanted to know if you use v4 already.

Btw. specific service versions can be found ind docker-compose. There is an image with a specific version. eg: https://github.com/ludeknovy/jtl-reporter/blob/main/docker-compose.yml#L5

**Reply by MoritzSalein** [reply]
I think you mean this entry:
`services:
  fe:
    image: novyl/jtl-reporter-fe:v4`

**Reply by ludeknovy** [reply]
yes, that's the image you use. I needed to know whether its v4 ;)

I will try to look into this issue soon.

**Reply by MoritzSalein** [reply]
Thank you for your support.

**Reply by ludeknovy** [reply]
I did a quick check and it works fine for me. Could you please try to upgrade to a newer version? 
![Screenshot 2022-01-28 at 14 39 30](https://user-images.githubusercontent.com/13610612/151556581-1db53be4-3be7-4417-97a3-090ef2f38779.png)
![Screenshot 2022-01-28 at 14 39 23](https://user-images.githubusercontent.com/13610612/151556585-efd66771-6308-4faa-8bb3-fa9fe74c00d9.png)

**Reply by MoritzSalein** [reply]
@ludeknovy 
With the latest version the dashboard shows the right value for network.
But now in the overview isn't the value of `network data`, there ist the value of` avg. latency`. Do you know why?
![2022-01-31 15_14_15-JtlReporter](https://user-images.githubusercontent.com/83296020/151809419-b58220e8-539d-43bb-a164-65f08b721a17.png)
![2022-01-31 15_12_01-JtlReporter](https://user-images.githubusercontent.com/83296020/151809422-5d3045b0-40f5-484b-80e9-c6748e7a3778.png)

**Reply by ludeknovy** [reply]
I believe that's because of the settings - navigate to project settings and there you can select which metrics get displayed.

**Reply by MoritzSalein** [reply]
Thank you, you're right. I have a different project setting in my new installation. I looked in the test run settings and found nothing to set it.  Therefore I asked you.
*5 stars for you support.*
